### PR TITLE
Adding functionality to get & delete incoming webhooks for APIv4

### PR DESCRIPTION
**Pull request by mpoornima**
#### Summary
APIs for GET & DELETE of incoming webhooks

#### Checklist
- [x] Added or updated unit tests (required for all new features)
- [x] Added API documentation ([mattermost/mattermost-api-reference#124](https://github.com/mattermost/mattermost-api-reference/pull/124))
- [x] All new/modified APIs include changes to the drivers

#### Notes to Reviewers
* I'm not completely sure about the permission checks that need to be implemented for these APIs, have followed similar to that of CREATE API. Do have a closer look at them. 

Thanks
Poornima

### CODE REVIEW

File: api4/webhook.go [hunk 1]
```diff
@@ -83,3 +87,79 @@ func getIncomingHooks(c *Context, w http.ResponseWriter, r *http.Request) {
...
+	} else {
+		channel, err = app.GetChannel(hook.ChannelId)
+		if err != nil {
+			c.Err = err
+			return
+		}
+
+		fmt.Println("session has permission to team ", app.SessionHasPermissionToTeam(c.Session, hook.TeamId, model.PERMISSION_MANAGE_WEBHOOKS))
+		fmt.Println("Channel open ", channel.Type)
+		fmt.Println("session has permission to channel ", app.SessionHasPermissionToChannel(c.Session, hook.ChannelId, model.PERMISSION_MANAGE_WEBHOOKS))
```
Comment (jwilander) [inline api4/webhook.go]: I think this is debug prints you forgot to remove :P

File: api4/webhook.go [hunk 2]
```diff
@@ -83,3 +87,79 @@ func getIncomingHooks(c *Context, w http.ResponseWriter, r *http.Request) {
...
+
+		if !app.SessionHasPermissionToTeam(c.Session, hook.TeamId, model.PERMISSION_MANAGE_WEBHOOKS) ||
+			(channel.Type != model.CHANNEL_OPEN && !app.SessionHasPermissionToChannel(c.Session, hook.ChannelId, model.PERMISSION_MANAGE_WEBHOOKS)) {
```
Comment (jwilander) [inline api4/webhook.go]: I think the permission check for the channel bit should be for `model.PERMISSION_READ_CHANNEL`. The team permission check will already check if they have permission to manage webhooks, then you just need to check if they are in the channel if it's not open with the second one

Does that make sense?

File: store/sql_webhook_store.go [hunk 3]
```diff
@@ -155,6 +159,9 @@ func (s SqlWebhookStore) GetIncoming(id string, allowFromCache bool) StoreChanne
 
 		if err := s.GetReplica().SelectOne(&webhook, "SELECT * FROM IncomingWebhooks WHERE Id = :Id AND DeleteAt = 0", map[string]interface{}{"Id": id}); err != nil {
 			result.Err = model.NewLocAppError("SqlWebhookStore.GetIncoming", "store.sql_webhooks.get_incoming.app_error", nil, "id="+id+", err="+err.Error())
+			if reflect.TypeOf(err) == reflect.TypeOf(sql.ErrNoRows) {
```
Comment (jwilander) [inline store/sql_webhook_store.go]: I think you can just check `err == sql.ErrNoRows` instead of using reflect

**Comment by mpoornima** [comment]
@jwilander Have made the requested changes. Please have a look.

**Pull request merged**